src/Sentry/Internal/Hub.cs

```diff
@@ -109,41 +108,28 @@ public void WithScope(Action<Scope> scopeCallback)
         {
             var transaction = new Transaction(this, context);
 
-            // Transactions are not handled by event processors, so some things need to be added manually
-
-            // Make a sampling decision if it hasn't been made already.
-            // It could have been made by this point if the transaction was started
-            // from a trace header which contains a sampling decision.
-            if (transaction.IsSampled is null)
+            // Dynamic sampling is ran regardless of whether a decision
+            // has already been made, as it can be used to override it.
+            if (_options.TracesSampler is { } tracesSampler)
             {
                 var samplingContext = new TransactionSamplingContext(
                     context,
                     customSamplingContext
                 );
 
-                var sampleRate =
-                    // Custom sampler may not exist or may return null, in which case we fallback
-                    // to the static sample rate.
-                    _options.TracesSampler?.Invoke(samplingContext)
-                    ?? _options.TracesSampleRate;
-
-                transaction.IsSampled = sampleRate switch
+                if (tracesSampler(samplingContext) is {} sampleRate)
```

Comment:
nit: Don't really care if it's `{}` or `{ }` but I wish a linter would just keep it consistent across the code base.

On line 113 you used `{ }` so:
```suggestion
                if (tracesSampler(samplingContext) is { } sampleRate)
```